app/src/main/java/com/example/androidacademyhw/details/FragmentMoviesDetails.kt

```diff
@@ -52,17 +57,33 @@ class FragmentMoviesDetails : Fragment()  {
         val movie = arguments?.getSerializable(Tags.KEY_MOVIE) as Movie
 
         with(binding) {
-            age.text = getString(R.string.details_item_text_pg, movie.pg)
-            movieNameText.text = movie.name
-            genre.text = movie.tags
-            ratingbar.rating = movie.rating.toFloat()
-            reviewAmount.text = getString(R.string.details_text_reviews, movie.reviewCount)
-            movieDescription.text = movie.storyLine
+
+            Glide.with(root.context)
+                .load(movie.poster)
+                .into(movieLogoImage)
+
+            age.text = getString(R.string.details_item_text_pg, movie.minimumAge)
+            movieNameText.text = movie.title
+            genre.text = movie.genres.map{it.name}.joinToString()
+            ratingbar.rating = movie.ratings.toFloat()
+            reviewAmount.text = getString(R.string.details_text_reviews, movie.numberOfRatings)
+            movieDescription.text = movie.overview
             initAdapter(movie.actors)
         }
 
         binding.buttonBackImage.setOnClickListener { navigateBack() }
         binding.buttonBackText.setOnClickListener { navigateBack() }
+
+        GlobalScope.launch(Dispatchers.Main) {
```

Comment:
We should not use GlobalScope here and anywhere in general. It breaks structured concurrency

Reply:
Done